context/app/static/js/components/detailPage/BulkDataTransfer/usePanelSet.tsx

```diff
@@ -251,19 +224,19 @@ function getGlobusPanel(status: number | undefined, panel: PanelSet, isLoading:
   return panel;
 }
 
-export const usePanelSet: () => PanelSet = () => {
+export const usePanelSet: (uuid: string, dbGapStudyURL?: string, accessType?: string) => PanelSet = (
+  uuid,
+  dbGapStudyURL,
+  accessType,
+) => {
   const { isAuthenticated, isHubmapUser } = useAppContext();
 
-  const {
-    entity: { dbgap_study_url, mapped_data_access_level: accessType, uuid },
-  } = useFlaskDataContext();
-
-  const hasDbGaPStudyURL = Boolean(dbgap_study_url);
+  const hasDbGaPStudyURL = Boolean(dbGapStudyURL);
   const { status: globusURLStatus, isLoading: globusURLIsLoading } = useFetchProtectedFile(uuid);
   const hasNoAccess = globusURLStatus === 403;
   const isNonConsortium = !isHubmapUser;
 
-  if (accessType !== 'Protected') {
+  if (!accessType || accessType !== 'Protected') {
```

Comment:
Are there situations we don't expect `accessType` or is this just for type safety?

Reply:
This was added to handle the initial loading state for `accessType`; since this was previously provided by flask data, it was never expected to be undefined, but since each descendant's bulk data transfer logic now gets handled independently, this is now loaded clientside for any processed descendants.